### ISSUE TITLE
Unable to use UPGRADE in RELAY2 stacks

### ISSUE THREAD

**Issue by pruivo** [jgroups-extras/RollingUpgrades#18]
Using `UPGRADE` with `RELAY2` stack throws a `ClassCastException: class org.jgroups.util.UUID cannot be cast to class org.jgroups.protocols.relay.SiteAddress`

```
WARNING: exception from server: io.grpc.StatusRuntimeException: CANCELLED: Failed to read message. (java.lang.ClassCastException: class org.jgroups.util.UUID cannot be cast to class org.jgroups.protocols.relay.SiteAddress (org.jgroups.util.UUID and org.jgroups.protocols.relay.SiteAddress are in unnamed module of loader 'app'))
```

This branch [1] (diff: [2]) contains the changes I've made.
Steps to reproduce:

* compile [1]
* run `./jchannel-test-4.x.sh`
* run `./jchannel-test-5.sh`
* run `./upgrade-server.sh`
* activate `UPGRADE`: `<probe> -bind_addr 127.0.0.1 -cluster xsite op=UPGRADE.activate[]`

[1] https://github.com/pruivo/RollingUpgrades/tree/t_test_relay
[2] https://github.com/pruivo/RollingUpgrades/commit/3bd39774dad0da3b4839157e1c7c2d1b90268322

**Comment by belaban** [comment]
@pruivo why is RELAY2 _below_ UPGRADE? IMO, it should be at the top of the stack! Will take a look...

**Comment by belaban** [comment]
Let's add lon.xml, nyc.xml and bridge.xml to all 3 configurations, so we can run RelayDemo (perhaps a script in ./bin as well).

**Comment by belaban** [comment]
@pruivo Note that I'm currently working on issue #14, to reduce code

**Comment by pruivo** [comment]
> @pruivo why is RELAY2 _below_ UPGRADE? IMO, it should be at the top of the stack! Will take a look...

Where?
The local cluster configuration (`config.xml`) does not have `UPGRADE`.
The bridge cluster (`bridge.xml`, used by `RELAY2`) has the `UPGRADE` at the top of the stack.

**Comment by belaban** [comment]
Sorry, it's more complex than that, let me think about it!

**Comment by belaban** [comment]
The local cluster does *not* need to have upgrade in it, only the bridge cluster!

**Comment by pruivo** [comment]
Another concern about the Infinispan PoC/Demo is `RELAY2` sending messages with `RelayHeader` and the `UPGRADE` protocol will "lose" this `Header`.

**Comment by belaban** [comment]
OK, I guess we'll have to prototype a generic 'header' in protobuf, and then provide selected implementations. Let's compile a list of issues and discuss them Monday. I guess we'll also have to support SiteUUID...

**Comment by belaban** [comment]
Probably fixed by #20

**Comment by belaban** [comment]
Fixed by #19
